app/views/products/index.html.erb

```diff
@@ -0,0 +1,37 @@
+<h1>Products</h1>
+
+<%= link_to 'New Product', new_product_path, class: 'btn btn-primary' %>
+
+<% if current_user %>
+  <div class="user-settings">
+    <h2>User Settings</h2>
+    <p><strong>Welcome, <%= current_user.username %></strong></p>
+    <%= link_to 'Edit Profile', edit_user_path(current_user), class: 'btn btn-secondary' %>
+    <%= link_to 'Delete Account', delete_user_path(current_user), method: :get, class: 'btn btn-danger', data: { confirm: 'Are you sure you want to delete your account?' } %>
+    <%= link_to 'Sign Out', destroy_session_users_path, method: :delete, data: { confirm: 'Are you sure you want to sign out?' } %>
+
+  </div>
+<% end %>
+
+<table>
+  <tr>
+    <th>Title</th>
+    <th>Description</th>
+    <th>Price</th>
+    <th>Stock</th>
+    <th>Actions</th>
+  </tr>
+  <% @products.each do |product| %>
+    <tr>
+      <td><%= product.title %></td>
+      <td><%= product.description %></td>
+      <td><%= product.price %></td>
+      <td><%= product.stock %></td>
+      <td>
+        <%= link_to 'Show', product_path(product), class: 'btn btn-info' %>
+        <%= link_to 'Edit', edit_product_path(product), class: 'btn btn-warning' %>
+        <%= link_to 'Delete', delete_product_path(product), method: :get, class: 'btn btn-danger' %>
+      </td>
+    </tr>
+  <% end %>
```

Comment:
may be we can add a text some thing like "no products available" when we don't have any results from DB rather than showing a table without any body!

Reply:
Okay bhai. I am adding it.